### PR TITLE
Added support for more library types using the prexisting nixpkgs machinery

### PR DESCRIPTION
**Pull request by asfishman**
Looks like I was able to implement a fix for this. I am not sure this is an idiomatic approach as I am relatively new tor rust. I appreciate any feedback.

### CODE REVIEW

File: src/resolve.rs [hunk 1]
```diff
@@ -95,6 +98,28 @@ impl CrateDerivation {
             .chain(metadata.workspace_members.iter())
             .any(|pkg_id| *pkg_id == package.id);
 
+        let crate_type = package
+            .targets
+            .iter()
+            .fold(Vec::<String>::new(), |types_vec, target| {
```
Comment (kolloch) [inline src/resolve.rs]: you probably want a `filter/collect` combination instead.

Comment (asfishman) [reply src/resolve.rs]: I will look into filter and collect.

For more info take a look at my associated issue: https://github.com/kolloch/crate2nix/issues/31

**Comment by kolloch** [comment]
Closing for now. Feel free to follow up.

**Comment by asfishman** [comment]
I don't have the bandwidth to work on this. Feel free to close.

**Pull request closed without merging**